File: houseRobber.js
class Solution {
  /**
   * @param {number[]} nums
   * @return {number}
   */
  rob(nums) {
    let prevMax = 0;
    let prevPrevMax = 0;
    let trueMax;

    for (let i = 0; i < nums.length; i++) {
      trueMax = Math.max(nums[i] + prevMax, prevPrevMax);
      prevMax = prevPrevMax;
      prevPrevMax = trueMax;
    }

    return trueMax;
  }
}

File: house_robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        prev_max = 0
        prev_prev_max = 0

        for i in range(len(nums)):
            true_max = max(nums[i] + prev_max, prev_prev_max)
            prev_max = prev_prev_max
            prev_prev_max = true_max

        return prev_prev_max
